Add prose to clarify that the invalid HTML is intentional

dipikabh opened the pull request:
<!-- 🙌 Thanks for contributing to MDN Web Docs. Adding details below will help us to merge your PR faster. -->

### Description

This PR updates the [Unordered lists](https://developer.mozilla.org/en-US/docs/Learn_web_development/Core/Structuring_content/Lists#unordered_lists) section in the Learn area to address the concern raised in https://github.com/mdn/content/issues/39611.

It clarifies that the first example is intentionally invalid and that's because it is part of a step-by-step explanation to show how to create unordered lists.

### Motivation

To avoid confusion about the code being incorrect and to help readers who might just be skimming the page.

dipikabh committed:
Update index.md

wbamberg commented:
Sorry to drive-by and feel free to ignore, but wouldn't it be OK just to have one step here? It doesn't seem that complicated a thing, to have to break it up (and then have to add the extra scaffolding of explaining that the middle step in invalid).

Josh-Cena commented:
Also does this intend to fix https://github.com/mdn/content/issues/39611?

bsmth reviewed (approved):
Nice thanks!

bsmth committed:
Merge branch 'main' into dipikabh-patch-1

dipikabh committed:
fix review comments

dipikabh commented:
Thanks everyone for your input.
I agree - this can be simplified a lot. And in this instance, we don't really need to show the intermediate step (the invalid HTML).
I've updated the prose: retained a bit of the steps phrasing and removed the incomplete HTML.
